Cypress + TypeScript 3.6.2 errors due to outdated JQuery types

<!-- Is this a question? Don't open an issue. Ask in our chat https://on.cypress.io/chat  -->

### Current behavior:

```
node_modules/cypress/types/jquery/index.d.ts:8155:87 - error TS2344: Type '"button" | "view" | "screenX" | "screenY" | "altKey" | "bubbles" | "cancelable" | "changedTouc
hes" | "ctrlKey" | "detail" | "eventPhase" | "metaKey" | "pageX" | "pageY" | "shiftKey" | ... 13 more ... | "touches"' does not satisfy the constraint '"repeat" | "butto
n" | "code" | "view" | "location" | "screenX" | "screenY" | "y" | "altKey" | "bubbles" | "cancelable" | "changedTouches" | "ctrlKey" | "detail" | "eventPhase" | ... 55 m
ore ... | "DOM_KEY_LOCATION_STANDARD"'.
  Type '"toElement"' is not assignable to type '"repeat" | "button" | "code" | "view" | "location" | "screenX" | "screenY" | "y" | "altKey" | "bubbles" | "cancelable" |
"changedTouches" | "ctrlKey" | "detail" | "eventPhase" | ... 55 more ... | "DOM_KEY_LOCATION_STANDARD"'.

8155         TData = null> extends Partial<Pick<PointerEvent & KeyboardEvent & TouchEvent, 'altKey' | 'bubbles' | 'cancelable' |
                                                                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
8156         'changedTouches' | 'ctrlKey' | 'detail' | 'eventPhase' | 'metaKey' | 'pageX' | 'pageY' | 'shiftKey' | 'view' |
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
8157         'char' | 'charCode' | 'key' | 'keyCode' | 'button' | 'buttons' | 'clientX' | 'clientY' | 'offsetX' | 'offsetY' |
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
8158         'pointerId' | 'pointerType' | 'screenX' | 'screenY' | 'targetTouches' | 'toElement' | 'touches'>> {
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

In particular:
> Type '"toElement"' is not assignable to type ...

### Desired behavior:

No TypeScript errors.

### Steps to reproduce: (app code and test code)

<!-- Issues without reproducible steps WILL BE CLOSED -->

<!-- You can fork https://github.com/cypress-io/cypress-test-tiny repo, set up a failing test, then tell us the repo/branch to try. -->
I have a few "real" projects exhibiting this issue, but here's one:
https://github.com/wKovacs64/pwl/pull/88

I can try to craft a more minimal reproduction as time permits.

### Versions

Cypress: 3.4.1
TypeScript: 3.6.2

### Possible fix:

https://github.com/cypress-io/cypress/blob/c912e095365bb857f24bfda791d5787b6da09e79/cli/package.json#L66

Bumping `@types/jquery` from `3.3.6` to `3.3.31` (the latest version at this time) should resolve the issue (technically, `3.3.23` was the first version that worked, but may as well go for the latest?). It worked if I manually replaced the contents of `<my_project>/node_modules/cypress/types/jquery` with the `@types/jquery@3.3.31` files, although I was not able to get `dtslint` to pass in the Cypress project itself due to conflicts somewhere:

```
Error: Errors in typescript@next for external dependencies:
../node_modules/@types/jquery/JQuery.d.ts(12941,5): error TS2375: Duplicate number index signature.
../node_modules/@types/jquery/legacy.d.ts(201,5): error TS2374: Duplicate string index signature.
../node_modules/@types/jquery/misc.d.ts(5,1): error TS6200: Definitions of the following identifiers conflict with those in another file: TypeOrArray, Node, htmlString, Selector, SuccessTextStatus, ErrorTextStatus, TextStatus, SuccessCallback, ErrorCallback, CompleteCallback, StatusCodeCallbacks, CSSHook, CallbackBase, Callback, Duration, Tweener, PropHook, EasingMethod, AnimationHook, Queue, QueueFunction, SpeedSettings, EventHandlerBase, EventHandler, TypeEventHandler, _TypeEventHandlers, SpecialEventHook, CoordinatesPartial, ValHook, _Falsy, jQuery, $, _Event, _UIEvent, _MouseEvent, _DragEvent, _KeyboardEvent, _TouchEvent, _FocusEvent
../node_modules/@types/jquery/misc.d.ts(24,9): error TS2374: Duplicate string index signature.
../node_modules/@types/jquery/misc.d.ts(1140,9): error TS2374: Duplicate string index signature.
../node_modules/@types/jquery/misc.d.ts(3788,9): error TS2374: Duplicate string index signature.
../node_modules/@types/jquery/misc.d.ts(3799,9): error TS2374: Duplicate string index signature.
../node_modules/@types/jquery/misc.d.ts(6405,9): error TS2374: Duplicate string index signature.
../node_modules/@types/jquery/misc.d.ts(6432,9): error TS2374: Duplicate string index signature.
../node_modules/@types/jquery/misc.d.ts(6550,9): error TS2374: Duplicate string index signature.
../node_modules/@types/jquery/misc.d.ts(6625,9): error TS2374: Duplicate string index signature.
```

Looks like the [same thing](https://circleci.com/gh/cypress-io/cypress/87417) the automated PR (#3343) ran into back in March.

### Additional information:

- `toElement` is an old Internet Explorer property. The DOM types that ship with TypeScript no longer include it in `lib.dom.d.ts`. See [this comment](https://github.com/microsoft/TypeScript/issues/32751#issuecomment-519292604). I think it was removed in [microsoft/TypeScript#32578](https://github.com/microsoft/TypeScript/pull/32578/files#diff-46fd87925e4552c166ec188712741c3fL10349). The version of `@types/jquery` (3.3.6) currently shipped with Cypress still use this property.

- The `@types/jquery` package went through a restructure in 3.3.21. Prior to that, the `toElement` key in question was present on the [`JQuery.Event<...>` interface in `index.d.ts`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/11ce407429b716983fc99c87a92f65f9c6133295/types/jquery/index.d.ts#L8160). The new definition of [`JQuery.Event`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/25731aaec086c61117b4bd66b38cf034453db013/types/jquery/misc.d.ts#L4053) now lives in [`misc.d.ts`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/25731aaec086c61117b4bd66b38cf034453db013/types/jquery/misc.d.ts) along with some new related types ([`TriggeredEvent`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/25731aaec086c61117b4bd66b38cf034453db013/types/jquery/misc.d.ts#L4528), [`EventBase`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/25731aaec086c61117b4bd66b38cf034453db013/types/jquery/misc.d.ts#L4752), etc.). None of these extend the same list of event types as the old version, and since none of them reference the now-absent `toElement`, the problem is resolved.

- Since `@types/jquery` 3.3.22 does not work but 3.3.23 does, I narrowed down the commits between those version release dates to c6fe757c..bf2beee3 and I'm pretty sure it was fixed in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/30144.

We are seeing this exact same thing.
cypress@3.4.1
typescript@3.6.2

```
../node_modules/cypress/types/jquery/index.d.ts:8155:87 - error TS2344: Type '"button" | "view" | "altKey" | "bubbles" | "cancelable" | "changedTouches" | "ctrlKey" | "detail" | "eventPhase" | "metaKey" | "pageX" | "pageY" | "shiftKey" | "char" | "charCode" | ... 13 more ... | "touches"' does not satisfy the constraint '"repeat" | "button" | "code" | "view" | "y" | "altKey" | "bubbles" | "cancelable" | "changedTouches" | "ctrlKey" | "detail" | "eventPhase" | "metaKey" | "pageX" | "pageY" | "shiftKey" | ... 54 more ... | "DOM_KEY_LOCATION_STANDARD"'.
  Type '"toElement"' is not assignable to type '"repeat" | "button" | "code" | "view" | "y" | "altKey" | "bubbles" | "cancelable" | "changedTouches" | "ctrlKey" | "detail" | "eventPhase" | "metaKey" | "pageX" | "pageY" | "shiftKey" | ... 54 more ... | "DOM_KEY_LOCATION_STANDARD"'.

8155         TData = null> extends Partial<Pick<PointerEvent & KeyboardEvent & TouchEvent, 'altKey' | 'bubbles' | 'cancelable' |
                                                                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
8156         'changedTouches' | 'ctrlKey' | 'detail' | 'eventPhase' | 'metaKey' | 'pageX' | 'pageY' | 'shiftKey' | 'view' |
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
8157         'char' | 'charCode' | 'key' | 'keyCode' | 'button' | 'buttons' | 'clientX' | 'clientY' | 'offsetX' | 'offsetY' |
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
8158         'pointerId' | 'pointerType' | 'screenX' | 'screenY' | 'targetTouches' | 'toElement' | 'touches'>> {
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/cli/package.json
+++ b/cli/package.json
@@ -63,7 +63,7 @@
     "@types/bluebird": "3.5.18",
     "@types/chai": "4.0.8",
     "@types/chai-jquery": "1.1.38",
-    "@types/jquery": "3.3.6",
+    "@types/jquery": "3.3.31",
     "@types/lodash": "4.14.122",
     "@types/minimatch": "3.0.3",
     "@types/mocha": "5.2.7",

--- a/cli/scripts/post-install.js
+++ b/cli/scripts/post-install.js
@@ -20,6 +20,12 @@ includeTypes.forEach((folder) => {
   shell.cp('-R', source, 'types')
 })
 
+// jQuery v3.3.x includes "dist" folder that just references back to itself
+// causing dtslint to think there are double definitions. Remove that folder.
+const typesJqueryDistFolder = join('types', 'jquery', 'dist')
+
+shell.rm('-rf', typesJqueryDistFolder)
+
 // fix paths to Chai, jQuery and other types to be relative
 shell.sed(
   '-i',

